src/Modules/TasksModule/components/TaskList/TaskList.tsx

```diff
@@ -1,5 +1,328 @@
+        );
+
+        setTasks(data.data);
+        setTotalPages(data.totalNumberOfPages);
+      } catch (err) {
+        handleApiError(err);
+      }
+      setIsLoading(false);
+    },
+    [pageNumber]
+  );
+
+  const onDeleteSubmit = async () => {
+    handleCloseDelete();
+    try {
+      await axios.delete(`${baseUrl}/Task/${taskId}`, {
+        headers: getRequestHeaders(),
+      });
+      getTask("", 10);
+      toast.success(`Deleted ${taskName} Successfully`);
+    } catch (err) {
+      handleApiError(err);
+    }
+  };
+
+  const handleCloseDelete = () => {
+    // resetting the values to default after closing the modal
+    setShowDelete(false);
+    setTaskId(0);
+    setTaskName("");
+  };
+
+  const handleShowDelete = (id: number, title: string) => {
+    // set the values to handle them in the delete process
+    setTaskId(id);
+    setTaskName(title);
+    setShowDelete(true);
+  };
+
+  const getTaskStatus = (status: string) => {
+    switch (status) {
+      case "ToDo":
+        return <p className={`${style.status} ${style.todo}`}>{status}</p>;
+      case "InProgress":
+        return (
+          <p className={`${style.status} ${style.inProgress}`}>{status}</p>
+        );
+      case "Done":
+        return <p className={`${style.status} ${style.done}`}>{status}</p>;
+      default:
+        return <p className={`${style.status}`}>Unknown</p>;
+    }
+  };
+
+  useEffect(() => {
+    getTask(taskTitle, 10);
+  }, [taskTitle, pageNumber, getTask]);
+
+  return (
+    <>
+      <section>
+        <div
+          className={`project-head container-fluid shadow-sm head-bg pt-5 pb-4 px-5`}
+        >
+          <div className={`row`}>
+            <div className="col-md-6">
+              <div>
+                <h2
+                  className={`text-theme text-lg-start text-sm-center text-center`}
+                >
+                  Tasks
+                </h2>
+              </div>
+            </div>
+            <div className={`col-md-6 text-lg-end text-sm-center text-center`}>
+              <div>
+                <div className={``}>
+                  <button
+                    onClick={() => {
+                      navigate("/dashboard/tasks-data");
+                    }}
+                    className={`main-btn`}
+                  >
+                    Add New Task
+                  </button>
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+
+        {/* Filtaration */}
+        <div
+          className={`filtaration container rounded-4 w-100 mt-3 ${style.inputSearch}`}
+        >
+          <div className="container">
+            <div className="input-container w-25">
+              <input
+                placeholder="Search By Name"
+                className={`input-field input-theme`}
+                type="text"
+                onChange={(e) => {
+                  setTaskTitle(e.target.value);
+                  getTask(taskTitle, 10);
```

Comment:
I haven't test the UI yet, but the setter function work asynchronously, this means that taskTitle that will be used here is the previous one, not the last one.